cpp/CMakeLists.txt

```diff
@@ -733,6 +733,19 @@ endif()
 
 set(ARROW_TEST_LINK_LIBS ${ARROW_MIN_TEST_LIBS})
 
+if (ARROW_BUILD_TESTS OR ARROW_BUILD_BENCHMARKS)
+  # TODO(wesm): maybe find a better place to put shared auxiliary testing tools
+  # that depend on gtest
+  add_library(arrow_testing_aux STATIC src/arrow/test-util.cc)
```

Comment:
Is there a reason why this must be `STATIC`?  Maybe using `OBJECT` or `SHARED` will solve your problem on Windows, probably `OBJECT` is what you're looking for: compile once, link "statically" into the tests everywhere.